src/components/hoc/withMeeting.jsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import {concatMap} from 'rxjs/operators';
+import PropTypes from 'prop-types';
+import {AdapterContext} from '../hooks/contexts';
+
+/**
+ * withMeeting is a higher-order component that takes a component (WrappedComponent)
+ * and returns a new component (WithMeeting). The new component needs to be provided with a meeting destination
+ * on the "meetingDestination" prop. The wrapped component will receive the meeting object on the "meeting" prop.
+ *
+ * Example:
+ *
+ * class OriginalComponent extends Component {
+ *   render(props) {
+ *     return <div>Meeting id is {props.meeting.ID}</div>;
+ *   }
+ * }
+ *
+ * const EnhancedComponent = withMeeting(OriginalComponent);
+ *
+ * <EnhancedComponent meetingDestination="<MEETING_DESTINATION>" />
+ *
+ * @param {React.Component} WrappedComponent The component to wrap and enhance
+ * @returns {React.Component} The wrapped component
+ */
+
+/**
+ * @param WrappedComponent
+ */
+export default function withMeeting(WrappedComponent) {
+  class WithMeeting extends React.Component {
+    constructor(props) {
+      super(props);
+      this.state = {
+        meeting: {},
+      };
+    }
+
+    componentDidMount() {
+      const {meetingsAdapter} = this.context;
+
+      this.meetingSubscription = meetingsAdapter
+        .createMeeting(this.props.meetingDestination)
+        .pipe(concatMap(({ID}) => meetingsAdapter.getMeeting(ID)))
```

Comment:
Do you really need to call and subscribe to `getMeeting`? Why would the HOC need to keep updated with the meeting object? Each wrapped component will already take care of that

Reply:
Hi @lalli-flores, I think yes. I do that because this is similar with the Hooks and this make sense. Another reason is that, in most cases, the enhanced component will need to have meeting status and that's why is required to make this call here.

Reply:
Reposting my reply:

> This I don't understand. Won't each wrapped component have access to the whole meeting object because it will do a subscription to getMeeting internally?

Reply:
Hm, maybe we're looking at different approaches here. One approach would be to wrap each component with this HOC, like `WebexMeeting`, `WebexMemberRoster` etc. However, the original intention was to wrap the webex meeting widget itself with the HOC (and likewise a dev-defined component that uses these webex components). This way, the widget (or the custom app) would have easy access to an up-to-date meeting object, from which they can extract:
- the meeting id, to provide it down to the webex components
- the meeting state, to decide how to display the components. For example, the widget could display a spinner while the meeting is being created (no meeting id yet), or it could decide to display the roster only after the meeting is JOINED.
- even the meeting controls could be further customized based on meeting state, for example, display an additional control when the user is sharing their screen.
If we just provide the meeting id, the dev (or the widget) would still need to subscribe to the `getMeeting()` adapter function to get meeting updates. And take care of the unsubscribing.
This "extra" subscription should be really cheap, as we are basically subscribing to a `BehaviorSubject`.
For easier access to the meeting id, we could also provide the wrapped component with a `meetingID` property, in addition to the (whole) meeting object.
If you still want us to only provide the meeting id to the wrapped component, we might need to do the same for the `useMeetingDestination` hook, for consistency reasons.

Reply:
Thanks for the thorough explanation @cipak, I appreciate the forward thinking!